src/main/scala/io/iohk/ethereum/crypto/ECDSASignature.scala

```diff
@@ -174,7 +174,7 @@ case class ECDSASignature(r: BigInt, s: BigInt, v: Byte) {
     import ECDSASignature.RLength
 
     def bigInt2Bytes(b: BigInt) =
-      ByteString(ByteUtils.bigIntToBytes(b, RLength))
+      ByteUtils.padLeft(ByteString(b.toByteArray).takeRight(RLength), RLength, 0)
```

Comment:
By the looks of it `ByteUtils.bigIntToBytes` _should_ have been doing the same thing, except it was lacking padding on the left. Unfortunately it doesn't have any docstrings or tests, so it's difficult to tell what it's supposed to be doing. I know that some private keys generated by Secp256k1 are 33 bytes and some are 32 when converted to a `BigInt`, which is why I had to change serialization [here](https://github.com/input-output-hk/mantis/commit/5ebf7c542b6b9f9d347f684969684c968e339ada#diff-38a246d4a03360047854e567b310975179c71d0fc410b7b1bd8af12fcaccb81bL67-L68). If this change made a difference in your case, does it mean you had a `BigInt` which had a different lenght? Do you have examples? Should `ByteUtils.bigIntToBytes` also do padding? Could you add unit tests?